cni/network/invoker_cns.go

```diff
@@ -229,20 +253,42 @@ func (invoker *CNSIPAMInvoker) Delete(address *net.IPNet, nwCfg *cni.NetworkConf
 		return errEmptyCNIArgs
 	}
 
-	req := cns.IPConfigRequest{
+	req := cns.IPConfigsRequest{
 		OrchestratorContext: orchestratorContext,
 		PodInterfaceID:      GetEndpointID(args),
 		InfraContainerID:    args.ContainerID,
 	}
 
-	if address != nil {
-		req.DesiredIPAddress = address.IP.String()
+	if len(addresses) > 0 {
+		req.DesiredIPAddresses = make([]string, len(addresses))
+		for i, ipAddress := range addresses {
+			req.DesiredIPAddresses[i] = ipAddress.IP.String()
+		}
 	} else {
 		log.Printf("CNS invoker called with empty IP address")
 	}
 
-	if err := invoker.cnsClient.ReleaseIPAddress(context.TODO(), req); err != nil {
-		return errors.Wrap(err, fmt.Sprintf("failed to release IP %v with err ", address)+"%w")
+	if err := invoker.cnsClient.ReleaseIPs(context.TODO(), req); err != nil {
+		// if we fail a release with a 404 error try using the old API
+		if errors.Is(err, cnscli.ErrAPINotFound) {
```

Comment:
In this case, what if the API fails a single time with 404 and we call the old API. ? Do we intend to call the old API for the subsequent calls ?
OR the old API is serving just as back up ? ( will probably produce the same results ?)

Reply:
yes, we do call old API if new API does not work